agent-connection/src/main/kotlin/com/luxoft/blockchainlab/corda/hyperledger/indy/AgentConnection.kt

```diff
@@ -77,29 +95,62 @@ class AgentConnection(val myAgentUrl: String, val invite: String? = null, val us
         val RESPONSE_RECEIVED = ADMIN_CONNECTIONS_BASE + "response_received"
 
         val STATE_REQUEST = ADMIN_BASE + "state_request"
+        val STATE_RESPONSE = ADMIN_BASE + "state"
     }
 
     private val webSocket = AgentWebSocketClient(URI(myAgentUrl))
 
+    fun checkState(stateMessage: State?, userName: String) : Boolean {
+        return if(stateMessage != null &&
+                stateMessage.content?.get("initialized") == true &&
+                stateMessage.content?.get("agent_name") == userName ) {
+            connectionStatus = ConnectionStatus.AGENT_CONNECTION_CONNECTED
+            true
+        } else {
+            connectionStatus = ConnectionStatus.AGENT_CONNECTION_DISCONNECTED
+            false
+        }
+    }
+
     init {
+        /**
+         * HTTP GET / in order to let the agent (python indy-agent) know its endpoint address
+         * indy-agent.py is incapable of determining its endpoint other than this way
+         */
+        val uri = URI(myAgentUrl)
+        val rootPath = "http://" + uri.host + ":" + uri.port + "/"
+        val rootUrl = URL(rootPath)
+        val res = rootUrl.openConnection().getInputStream()
```

Comment:
I still dont like idea, of any "magic" for agent specific implementation.
Plus, it is bad idea to keep connection opened for no real reason.

Reply:
Would you suggest another way of making agents to generate valid invites? Invites are core agent functionality, we can't simply omit it.

Reply:
I think than this is agents responsibility to say on which address it can be accedes, not clients problem. Thous I would just remove all this code)

Reply:
> You may add this "hack" to integration test on before handler.

Accepted. I'll fix this.